architecture/adr-list/adr-005-genesis-parameters.md

```diff
@@ -0,0 +1,165 @@
+
+> Cosmos v0.42.5 parameters are described.
+
+## Context
+
+Genesis consists of Tendermint consensus engine parameters and Cosmos app-specific parameters.
+
+### Consensus parameters
+
+Proposed values:
+
+* block
+  * max\_bytes = `22020096` \(~22MB\)
+    * Cosmos hub: `200000` \(~200KB\)
+  * max\_gas = `-1` \(no gas limit\)
+    * Cosmos hub: `2000000`
+  * time\_iota\_ms = `1000`
+    * Cosmos hub: `1000`
+    * **Deprecated, unused**
+* evidence
+  * max\_age\_num\_blocks = `100000`
+    * Max age of evidence, in blocks. The basic formula for calculating this is: `MaxAgeDuration / {average block time}`.
+  * max\_age\_duration = `172800000000000`
+    * Max age of evidence, in time. It should correspond with an app's "unbonding period".
+  * max\_bytes = `1048576`
+    * This sets the maximum size of total evidence in bytes that can be committed in a single block and should fall comfortably under the max block bytes.
+* validator
+  * pub\_key\_types = `[ "ed25519" ]`
+
+[Here](https://docs.tendermint.com/master/tendermint-core/using-tendermint.html#genesis) you can find more about Tendermint genesis parameters.
+
+### Application parameters
+
+Cosmos application is divided into modules. Each module has parameters that help to adjust the module's behavior. Here are proposed values for default modules:
+
+* auth
+  * max\_memo\_characters = `512`
+    * Max number of characters in the memo field
+  * tx\_sig\_limit = `7`
+    * Max number of signatures
+  * tx\_size\_cost\_per\_byte = `10`
+    * Gas cost of transaction byte
+  * sig\_verify\_cost\_ed25519 = `590`
+    * Cost of `ed25519` signature verification
+  * sig\_verify\_cost\_secp256k1 = `1000`
+    * Cost of `secp256k1` signature verification
+* bank
+  * send\_enabled = `[]`
+    * Enables send for specific denominations
+  * default\_send\_enabled = `true`
+    * The default send enabled value allows send transfers for all coin denominations
+* crisis
+  * constant\_fee = `{ "denom": "cheq", "amount": "1000" }`
+    * The fee is used to verify the invariant in the crisis module. Rean more about invariants [here](https://docs.cosmos.network/v0.43/building-modules/invariants.html).
+* distribution
+  * community\_tax = `0.02`
+    * The percent of rewards that goes to the community fund pool
+  * base\_proposer\_reward = `0.01`
+    * Base reward that proposer gets
+  * bonus\_proposer\_reward = `0.04`
+    * Bonus reward that proposer gets which depends on the number of precommits included to the block
+  * withdraw\_addr\_enabled = `true`
+    * Whether withdrawal address can be changed or not. By default, it's the delegator's address.
+* evidence
+  * No parameters
+* genutil
+  * Used to manage initial transactions such as genesis validators creation
+* gov
+  * deposit\_params
+    * min\_deposit = `[{ "denom": "stake", "amount": "10000000" }]`
+      * The minimum deposit for a proposal to enter the voting period.
```

Comment:
Should/can the denom for this be cheq? Also, see the updated genesis parameters, proposed minimum deposit now at 8000 cheq.

Reply:
Sure